Handler/VerifyConfigurationTrait.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Copyright © OpenGento, All rights reserved.
+ * See LICENSE bundled with this library for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Opengento\Logger\Handler;
+
+use Exception;
+
+trait VerifyConfigurationTrait
+{
+    /**
+     * @param array $record
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function handle(array $record): void
+    {
+        if ($this->isEnabled()) {
+            $this->getInstance()->handle($record);
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    private function isEnabled(): bool
+    {
+        return $this->scopeConfig->getValue($this->isEnabled) === '1';
```

Comment:
Ton trait ne peut marcher que s'il est implémenté par des classes qui exposent `scopeConfig` et `isEnabled` donc. 

La déclaration de ces attributs devrait de ce fait descendre dans le trait ?

Reply:
Oui. Ni fait ni à faire. 

Il faut que "$this->isEnabled" passe en tant que paramètre dans la fonction, et que le scope config soit instancié via l'OM (car pas d'injection de dépendance via les traits si le parent a déjà un construct).

Même remraque pour au dessus le "->getInstance" devrait être passé en param (interface commune)